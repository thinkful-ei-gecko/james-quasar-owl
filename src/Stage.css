.on-stage {
  border: 3px solid red;
  flex: 5 1 auto;
  display: flex;
  flex-direction: column;
}


.stage-person {
border: 2px solid black;
flex: 1 1 auto;
color: white;
background-color: #353535;
}

.stage-person__avatar {
  border: 2px solid blue;
  width: 150px;
  height: 150px;
  border-radius: 100%;
  display: flex;
  justify-content: center;
  background-color: #060606;

}


.stage {
  display: flex;
  height: 100%;
  flex-wrap: wrap;
  
}