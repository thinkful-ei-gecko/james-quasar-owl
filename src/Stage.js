import React from 'react';
import store from './store';

export default function Stage(props) {
  const peopleOnStage = store.participants.filter(person => {
    return person.onStage
  })
  return (
    <section className="on-stage">
      <header>
        {/* remove after styles appropriately show separate from List */}
        <h1>The Stage</h1>
      </header>
      {
        peopleOnStage.map( person => (
          <ul key={person.id}>
            <li>{person.name}</li>
            <li><img src={person.avatar} alt={person.name + ' avatar'}/></li>
          </ul>
        ))
      }
    </section>
  );
}
