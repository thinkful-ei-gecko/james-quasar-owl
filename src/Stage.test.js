import React from 'react';
import ReactDOM from 'react-dom';
import Stage from './Stage.js';
import renderer from 'react-test-renderer';

describe('Stage tests', () => {
  it('renders without breaking', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Stage key={1}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders same UI', () => {
    const tree = renderer.create(<Stage key={1} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});