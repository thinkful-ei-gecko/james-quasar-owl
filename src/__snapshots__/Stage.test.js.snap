// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stage tests renders same UI 1`] = `
<section
  className="on-stage"
>
  <header>
    <h1>
      The Stage
    </h1>
  </header>
  <ul>
    <li>
      Koren Templeton
    </li>
    <li>
      <img
        alt="Koren Templeton avatar"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
    </li>
  </ul>
  <ul>
    <li>
      Ashla Attwool
    </li>
    <li>
      <img
        alt="Ashla Attwool avatar"
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
    </li>
  </ul>
</section>
`;
